### ISSUE TITLE
[Feature] Low end hardware support

### ISSUE THREAD

**Issue by nokatt** [stashapp/stash#2595]
**Is your feature request related to a problem? Please describe.**
My hardware isn't able to transcode most videos in real time. Because the application lacks options to bypass this issue it becomes useless on low end hardware as there is no way of actually playing your organized files. 

**Describe the solution you'd like**
An option that enables you to click on a scene or a movie and instead of opening the web player it simply sends the URL to an external video player like VLC. I'm not sure whether this is possible with browsers, but this extension accomplishes something similar. https://github.com/andy-portmen/open-in-vlc

**Describe alternatives you've considered**
I tried [Stash_Helper](https://github.com/philpw99/Stash_Helper) and the [Kodi plugin](https://github.com/gitgiggety/plugin.video.stash). Both make Stash barely useable for me as I don't have to transcode the videos but the handling is not really ideal.

**Comment by nokatt** [comment]
> A stream link in the "File info" panel isn't a proper solution? Works fine for me, except lack of metadata in external player and playlists playback. But at least no server-side transcoding!

Just try to open 5 videos like that. It will probably take you like 20-30s just to open the videos. Now try to imagine handling thousands of videos.

**Comment by YurikaL** [comment]
> Just try to open 5 videos like that. It will probably take you like 20-30s just to open the videos.

How about [Open With](https://github.com/darktrojan/openwith) extension? It's a bit similar to Open in VLC, but also allows you to add arbitrary external players and display items in the context menu. With this extension you can open a single video in just two clicks: right click to stream link -> "Open in [pre-configured player]".

BTW, I suggest you to rename this issue to something like "Built-in external player support" for more clarification. Low end hardware is not only reason to using external player!

**Comment by nokatt** [comment]
> How about [Open With](https://github.com/darktrojan/openwith) extension? It's a bit similar to Open in VLC, but also allows you to add arbitrary external players and display items in the context menu. With this extension you can open a single video in just two clicks: right click to stream link -> "Open in [pre-configured player]".

It's basically the same as Open in VLC and therefore not ideal. It's not _just two clicks_ but rather a click followed by mouse movement followed by another click. This can lead to 100-1000ms delay and gets very annoying with thousands of videos. 

> BTW, I suggest you to rename this issue to something like "Built-in external player support" for more clarification. Low end hardware is not only reason to using external player!

I didn't choose that title since it's only one solution to the underlying problem and I'm not even sure if it can be implemented like that. Another solution to the issue would be an option to disable transcoding completely. I wouldn't mind having such an option since most of the videos of my collection play perfectly fine in the browser.  

The main issue I wanted to point out is that this application is nearly completely useless if you have lower end hardware and intend to use it to stream videos.

**Comment by goodder21** [comment]
Another possibility would be to add the option to pre-transcode the videos to specific formats in a transparent way. 
This would help in case:
- there is a slow internet connection to the server running stash
- the file size is big (e.g., 4K video)
- the server is too slow for live transcoding to a lower resolution

Of course the file generated should live in the "generated" folder and the video player should give information about which format is pre-transcoded. Better if this could be done also on a file by file basis (if storage space usage is a concern).

**Comment by nokatt** [comment]
> * there is a slow internet connection to the server running stash
> * the file size is big (e.g., 4K video)
> * the server is too slow for live transcoding to a lower resolution
 
It's running locally on my NAS and my CPU isn't just fast enough to transcode the video files. There aren't even huge file sizes even 480p videos are buffering all the time. 

> Another possibility would be to add the option to pre-transcode the videos to specific formats in a transparent way. This would help in case:
> Of course the file generated should live in the "generated" folder and the video player should give information about which format is pre-transcoded. Better if this could be done also on a file by file basis (if storage space usage is a concern).

Sorry I know you just tried to help but that proposal sounds stupid to me. I think least people on this planet would want to litter their storage with transcoded files just so they can view the videos in the browser even though most of the files play perfectly fine in the browser anyway. 


**Probably the easiest solution would be an option to disable transcoding completely and live with the fact that not all videos can be played in the browser.** 

But people don't seem to care about such an option and it's beyond me why. Does every user here just run their instances on expensive hardware or do they just have 240p files? Am I overlooking something? I would really like an application to organize my porn but the forced transcoding is just killing the application completely for me.

**Comment by MrX292** [comment]
You can try the Userscript Stash Open Media Player from https://github.com/7dJx1qP/stash-userscripts
Can you check what codec the files that are transcoding are in? 
Depending on the codec a different Browser may help.
You can check https://caniuse.com for what browser supports what.

**Comment by nokatt** [comment]
Uhm I should have checked Stash before making the comment. Somehow there is no issue with streaming anymore. I can even stream 4k videos without noticeable buffering and the CPU doesn't even sweat, it's sitting at 10% usage on all cores while streaming a 15GB file. I have no idea how this happened. I didn't change any settings and I didn't update the executable. I checked the CPU usage before I created the issue and it was similar to now at 3-5% at all cores.
